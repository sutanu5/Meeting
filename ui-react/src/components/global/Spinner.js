import React from 'react';

const Spinner = () => (
  <p id="fb__spinner-wrap">
    <img id="fb__spinner" alt="Loading..." src="/svgs/spinner.svg" />
  </p>
);

export default Spinner;
