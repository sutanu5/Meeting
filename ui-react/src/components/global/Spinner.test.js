import Spinner from './Spinner';

describe('Spinner component', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Spinner />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders fb__spinner-wrap', () => {
    const div = wrapper.find('#fb__spinner-wrap').exists();
    expect(div).toBeTruthy();
  });

  it('renders fb__spinner-wrap with img', () => {
    const div = wrapper.find('#fb__spinner-wrap').html();
    expect(div).not.toBeEmpty();
  })
});