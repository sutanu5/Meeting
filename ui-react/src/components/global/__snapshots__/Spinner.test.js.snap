// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component renders correctly 1`] = `
<p
  id="fb__spinner-wrap"
>
  <img
    alt="Loading..."
    id="fb__spinner"
    src="/svgs/spinner.svg"
  />
</p>
`;
